### DISCUSSION TITLE
Trigger Cronicle events through file events

### DISCUSSION THREAD

**Discussion by dheise-rheinenergie-trading-com** [jhuckaby/Cronicle#390] [Show and tell]
I have several requirements where newly created files in folders have to be proccessed as soon as possible. In the past, I created Cronicle events with a a very short timing (5 minutes or even 1 minute). These leads to a lot of unnecessary runnings. To avoid these, I created a solution to trigger such events only if needed.

So it works:

1. Create a endless running event which polls the affected folders in a loop (based on a config file) 
2. Check if there are changes in this folder since the last check
3. If there are changes - check if there are newly created files to be searched for (the pattern is set in the same config file above)
4. Trigger the revelant event in Cronicle through a Rest API call (the event id set in the same config file above)

This way I can extremely reduce the number of events executed and still react to and process new files within seconds.

If there is interest i could also provide the scripts for it.

**Comment by mikeTWC1984** [comment]
How do you control that this endless job is up and running?
Actually "service" like events could be great extra feature. I think many people use "every minute" cron schedule to emulate service, which is a bit ugly.

**Reply by dheise-rheinenergie-trading-com** [reply]
the endless job is a really simple 2-event setup:

**event 1 - the real job**
- timing is set to "on demand"
- concurrency is 1
- chain points to itself for failure exits

**event 2 - the stop job (sto prevent large log files in event 1)**
- timing is set to once per day
- send a stop command to the script which was started by event 1
- chain points to event 1

**Reply by mikeTWC1984** [reply]
Yeah, self-chaining is a neat idea ) I'd think "do something on file upload" is pretty common scenario, so definitely go ahead and share your scripts. 
For scenarios when users upload files via some UI, I'll probably check if file sharing app supports webhooks. Good options for that are [minio](https://min.io) and [filebrowser](https://github.com/filebrowser/filebrowser)
